In [ ]:
import numpy as np
import pandas as pd

**Task-1**

1.

In [ ]:
np.arange(0,21,1)

array([ 0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 16,
       17, 18, 19, 20])

In [ ]:
arr=np.random.randint(10, 100, size=(4, 5))

In [ ]:
np.eye(3)

array([[1., 0., 0.],
       [0., 1., 0.],
       [0., 0., 1.]])

2.

In [ ]:
print(np.mean(arr))


50.2


In [ ]:
print(np.median(arr))


52.0


In [ ]:
print(np.std(arr))

27.086158826972863


**Task-2**



In [ ]:
#1
arr = np.arange(16)
reshaped_arr = arr.reshape(4, 4)
print(reshaped_arr)


[[ 0  1  2  3]
 [ 4  5  6  7]
 [ 8  9 10 11]
 [12 13 14 15]]


In [ ]:
#2
print(reshaped_arr+np.array([1,2,3,4]))

[[ 1  3  5  7]
 [ 5  7  9 11]
 [ 9 11 13 15]
 [13 15 17 19]]


In [ ]:
#3
flattened_arr = reshaped_arr.flatten()
print(flattened_arr)

[ 0  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15]


**Task-3**

In [ ]:
#1
data = {
    'Name': ['Alice', 'Bob', 'Charlie', 'Diana', 'Evan'],
    'Age': [25, 30, 28, 35, 40],
    'Department': ['IT', 'HR', 'IT', 'Marketing', 'HR'],
    'Salary': [50000, 45000, 72000, 58000, 60000]
}

df = pd.DataFrame(data)

In [ ]:
#2
df['Bonus'] = df['Salary'] * 0.10
hr_employees = df[df['Department'] == 'HR']
average_salary_by_dept = df.groupby('Department')['Salary'].mean()

df.to_csv('employees.csv', index=False)
df, hr_employees, average_salary_by_dept

(      Name  Age Department  Salary   Bonus
 0    Alice   25         IT   50000  5000.0
 1      Bob   30         HR   45000  4500.0
 2  Charlie   28         IT   72000  7200.0
 3    Diana   35  Marketing   58000  5800.0
 4     Evan   40         HR   60000  6000.0,
    Name  Age Department  Salary   Bonus
 1   Bob   30         HR   45000  4500.0
 4  Evan   40         HR   60000  6000.0,
 Department
 HR           52500.0
 IT           61000.0
 Marketing    58000.0
 Name: Salary, dtype: float64)

**Task-4**

In [ ]:
#1
df=pd.read_csv("/content/Mental_Health_and_Social_Media_Balance_Dataset.csv")

In [ ]:
#2
df.head(5)

User_ID  Age  Gender  Daily_Screen_Time(hrs)  Sleep_Quality(1-10)  \
0    U001   44    Male                     3.1                  7.0   
1    U002   30   Other                     5.1                  7.0   
2    U003   23   Other                     7.4                  6.0   
3    U004   36  Female                     5.7                  7.0   
4    U005   34  Female                     7.0                  4.0   

   Stress_Level(1-10)  Days_Without_Social_Media  Exercise_Frequency(week)  \
0                 6.0                        2.0                       5.0   
1                 8.0                        5.0                       3.0   
2                 7.0                        1.0                       3.0   
3                 8.0                        1.0                       1.0   
4                 7.0                        5.0                       1.0   

  Social_Media_Platform  Happiness_Index(1-10)  
0              Facebook                   10.0  
1              LinkedIn                   10.0  
2               YouTube                    6.0  
3                TikTok                    8.0  
4           X (Twitter)                    8.0

In [ ]:
df.tail(5)


User_ID  Age  Gender  Daily_Screen_Time(hrs)  Sleep_Quality(1-10)  \
495    U496   23    Male                     6.9                  5.0   
496    U497   43  Female                     5.6                  7.0   
497    U498   41    Male                     7.7                  5.0   
498    U499   23    Male                     4.2                  9.0   
499    U500   43  Female                     5.9                  5.0   

     Stress_Level(1-10)  Days_Without_Social_Media  Exercise_Frequency(week)  \
495                 7.0                        4.0                       2.0   
496                 6.0                        5.0                       2.0   
497                 7.0                        2.0                       2.0   
498                 7.0                        0.0                       2.0   
499                 8.0                        3.0                       3.0   

    Social_Media_Platform  Happiness_Index(1-10)  
495           X (Twitter)                   10.0  
496              Facebook                    9.0  
497              LinkedIn                    8.0  
498              Facebook                    9.0  
499           X (Twitter)                    7.0

In [ ]:
df.shape


(500, 10)

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 500 entries, 0 to 499
Data columns (total 10 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   User_ID                    500 non-null    object 
 1   Age                        500 non-null    int64  
 2   Gender                     500 non-null    object 
 3   Daily_Screen_Time(hrs)     500 non-null    float64
 4   Sleep_Quality(1-10)        500 non-null    float64
 5   Stress_Level(1-10)         500 non-null    float64
 6   Days_Without_Social_Media  500 non-null    float64
 7   Exercise_Frequency(week)   500 non-null    float64
 8   Social_Media_Platform      500 non-null    object 
 9   Happiness_Index(1-10)      500 non-null    float64
dtypes: float64(6), int64(1), object(3)
memory usage: 39.2+ KB


In [ ]:
#3
df.isnull().sum()

User_ID                      0
Age                          0
Gender                       0
Daily_Screen_Time(hrs)       0
Sleep_Quality(1-10)          0
Stress_Level(1-10)           0
Days_Without_Social_Media    0
Exercise_Frequency(week)     0
Social_Media_Platform        0
Happiness_Index(1-10)        0
dtype: int64

In [ ]:
#4
df.describe()

Age  Daily_Screen_Time(hrs)  Sleep_Quality(1-10)  \
count  500.000000              500.000000           500.000000   
mean    32.988000                5.530000             6.304000   
std      9.960637                1.734877             1.529792   
min     16.000000                1.000000             2.000000   
25%     24.000000                4.300000             5.000000   
50%     34.000000                5.600000             6.000000   
75%     41.000000                6.700000             7.000000   
max     49.000000               10.800000            10.000000   

       Stress_Level(1-10)  Days_Without_Social_Media  \
count          500.000000                 500.000000   
mean             6.618000                   3.134000   
std              1.542996                   1.858751   
min              2.000000                   0.000000   
25%              6.000000                   2.000000   
50%              7.000000                   3.000000   
75%              8.000000                   5.000000   
max             10.000000                   9.000000   

       Exercise_Frequency(week)  Happiness_Index(1-10)  
count                500.000000             500.000000  
mean                   2.448000               8.376000  
std                    1.428067               1.524228  
min                    0.000000               4.000000  
25%                    1.000000               7.000000  
50%                    2.000000               9.000000  
75%                    3.000000              10.000000  
max                    7.000000              10.000000

**Task-5**

In [ ]:
from sklearn.datasets import load_iris


In [ ]:

iris = load_iris()
df = pd.DataFrame(iris.data, columns=iris.feature_names)
df['target'] = iris.target

df['binary_target'] = (df['target'] == 1).astype(int)

df.head()


sepal length (cm)  sepal width (cm)  petal length (cm)  petal width (cm)  \
0                5.1               3.5                1.4               0.2   
1                4.9               3.0                1.4               0.2   
2                4.7               3.2                1.3               0.2   
3                4.6               3.1                1.5               0.2   
4                5.0               3.6                1.4               0.2   

   target  binary_target  
0       0              0  
1       0              0  
2       0              0  
3       0              0  
4       0              0

In [ ]:
from sklearn.model_selection import train_test_split

X = df[iris.feature_names]
y = df['binary_target']

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=42)

(X_train.shape, X_test.shape)


((105, 4), (45, 4))

In [ ]:
from sklearn.linear_model import LogisticRegression




In [ ]:
model = LogisticRegression(max_iter=200)
model.fit(X_train, y_train)

LogisticRegression(max_iter=200)

In [ ]:
y_pred = model.predict(X_test)
y_pred[:5]


array([1, 0, 1, 0, 0])

In [ ]:
from sklearn.metrics import confusion_matrix, accuracy_score, precision_score, recall_score, f1_score

conf_matrix = confusion_matrix(y_test, y_pred)

accuracy = accuracy_score(y_test, y_pred)

precision = precision_score(y_test, y_pred)

recall = recall_score(y_test, y_pred)

f1 = f1_score(y_test, y_pred)

conf_matrix, accuracy, precision, recall, f1


(array([[29,  3],
        [ 9,  4]]),
 0.7333333333333333,
 0.5714285714285714,
 0.3076923076923077,
 0.4)

**Confusion Matrix:**
The confusion matrix is a 2x2 matrix:
[[29, 3],
[ 9, 4]]






*   **True Negatives (TN)** = 29: Correctly predicted "not Versicolor".
*   **False Positives (FP)** = 3: Incorrectly predicted "Versicolor".
*   **False Negatives (FN)** = 9: Incorrectly predicted "not Versicolor".
*   **True Positives (TP)** = 4: Correctly predicted "Versicolor".








**Metrics:**
- **Accuracy**: 73.33% — Percentage of correct predictions out of all predictions.
- **Precision**: 57.14% — How many of the predicted Versicolor cases were actually Versicolor.
- **Recall**: 30.77% — How many of the actual Versicolor cases were correctly identified.
- **F1-Score**: 40% — The harmonic mean of precision and recall, balancing the trade-off between them.

These metrics show that the model has a decent accuracy, but recall is much lower, indicating that the model misses many actual Versicolor instances.